internal/mage/sdk/nodejs.go

```diff
@@ -38,48 +39,64 @@ func (t NodeJS) Test(ctx context.Context) error {
 }
 
 // Generate re-generates the SDK API
-func (t NodeJS) Generate(ctx context.Context) error {
+func (t Nodejs) Generate(ctx context.Context) error {
 	panic("FIXME")
 }
 
-// Publish publishes the NodeJS SDK
-func (t NodeJS) Publish(ctx context.Context, tag string) error {
-	panic("FIXME")
+// Publish publishes the Node.js SDK
+func (t Nodejs) Publish(ctx context.Context, tag string) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	var (
+		// version = strings.TrimPrefix(tag, "sdk/nodejs/v")
+		tokenSecret = c.Host().EnvVariable("NPM_TOKEN").Secret()
+	)
+	token, err := tokenSecret.Plaintext(ctx)
+	if err != nil {
+		return err
+	}
+
+	if token == "" {
+		return errors.New("NPM_TOKEN environment variable must be set")
+	}
+
+	build := nodeJsBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"npm", "run", "build"},
+		})
+
+	args := []string{"npm", "publish"}
+
+	_, err = build.
+		WithSecretVariable("NPM_TOKEN", tokenSecret).
+		Exec(dagger.ContainerExecOpts{Args: args}).
+		ExitCode(ctx)
+
+	return err
 }
 
-// Bump the NodeJS SDK's Engine dependency
-func (t NodeJS) Bump(ctx context.Context, version string) error {
+// Bump the Node.js SDK's Engine dependency
+func (t Nodejs) Bump(ctx context.Context, version string) error {
 	panic("Andrea / Erik / Tom: https://github.com/dagger/dagger/pull/3783#issuecomment-1311833703")
 }
 
-func nodeJSBase(c *dagger.Client) *dagger.Container {
-	// FIXME: change to `util.Repository(c).Directory("sdk/python")` once #3459 is merged
+func nodeJsBase(c *dagger.Client) *dagger.Container {
+	src := c.Directory().WithDirectory("/", util.Repository(c).Directory("sdk/nodejs"))
 
-	src := c.Directory().WithDirectory("/", util.Repository(c))
+	base := c.Container().From("node:16-alpine")
 
-	base := c.Container().
-		// ⚠️  Keep this in sync with the engine version defined in package.json
-		From("node:16-alpine").
+	baseWithSrc := base.
+		WithMountedDirectory("/app", src).
 		WithWorkdir("/app")
 
-	base = base.WithFS(
```

Comment:
Not doing this will mean that any change in a `.js` file will re-run `yarn install`

Reply:
Got it! We should fix this in the Python SDK too @helderco